### ISSUE TITLE
No modal asking the user to enter the reason for changes when the Track Reason Changes form was killed while editing it

### ISSUE THREAD

**Issue by dbemke** [getodk/collect#5301]
#### ODK Collect version
v2021.3.4, v2022.1.0, the store version (2022.3.5), master version 31897362c00974e7df889ef685337244f0ac5604
#### Android version
10, 13
#### Device used
Redmi 9T, Pixel 6A
#### Problem description
After killing the Track Changes Reason form there is no modal asking the user to enter the reason for changes when the form was killed while editing it.
#### Steps to reproduce the problem
1. Go to "Fill Blank Form".
2. Go to Track Changes Reason form.
[track-changes-reason-on-edit.xml.txt](https://github.com/getodk/collect/files/9746996/track-changes-reason-on-edit.xml.txt)
3. Fill the first question.
4. Save and exit the form.
5. Go to "Edit Saved Form".
6. Open the filled form.
7. Add something in the filled answer.
8. Minimize and kill the app.
9. Go back to the same form in "Edit Saved Form".
10. Go to end,  save and exit the form.

#### Expected behavior
The user should be asked to enter a reason for changes. The `question` event with the changed answer should also be present (this is true in v2022.3.5 but not in other versions).

**Comment by seadowg** [comment]
@lognaturel I'm wondering if we want to make our implementation of `track-changes-reason` a little simpler to make it easier to catch edge cases like this. Right now, we only prompt for a change reason when saving if the form is being edited AND the user has made changes. If we're editing a form from a savepoint where changes were already made, we have no way of knowing (currently) if the savepoint has changes relative to the actual instance. 

We could definitely add in machinery to detect these changes through diffing actual instance XML/serialized objects for instance, but I'm wondering if we could just always prompt for a save reason when saving a form that is being edited? This would mean that you would sometimes see `change reason` events in audit logs where no changes were made if a user saves an edited form without changes - I'd hope showing the prompt would make it less likely that they would both going through the process though and would back out and "Ignore changes".

**Comment by seadowg** [comment]
I've found another (existing) way to bypass the change reason prompt:

1. Enable "Don't keep Activities" in Developer settings
1. Edit form with `track-changes-reason`
1. Change a value
1. Save
1. Minimize app while on change reason prompt
1. Return to app (this causes the app to crash)
1. Reopen app and edit form again and save (form saved successfully with new value and without change reason)

The crash is a separate issue, but everything should still work properly with change reason even with the crash. I reproduced this with the current `master` and `v2022.2.0` so it's existed for a while. For me, this is another vote for my above suggestion.
